web/src/engine/websites/OlympusScanlation.ts

```diff
@@ -1,36 +1,114 @@
+    id: number,
+    pages: string[]
+}
+
+type APIChapters = {
+    data: APIChapter[]
+}
+
+type APIPages = {
+    chapter: APIChapter
+}
+
+const mangaClipboardScript = `
+    new Promise(resolve => {
+        const path = window.location.pathname;
+        const mangaSlug = __NUXT__.data[path].data.slug;
+        const mangatype = __NUXT__.data[path].data.type;
+        const mangaName = __NUXT__.data[path].data.name.trim();
+        resolve({slug : mangaSlug, type : mangatype, name: mangaName });
+    });
+
+`;
 
-@Madara.MangaCSS(/^https?:\/\/olympusscanlation\.com\/manga\/[^/]+\/$/)
-@Madara.MangasMultiPageAJAX()
-@Madara.ChaptersSinglePageAJAXv1()
-@Madara.PagesSinglePageCSS()
 @Common.ImageAjax()
 export default class extends DecoratableMangaScraper {
 
+    private readonly apiUrl = 'https://dashboard.olympusv2.gg';
     public constructor() {
-        super('olympusscanlation', 'Olympus Scanlation', 'https://olympusscanlation.com'/*, Tags.Media., Tags.Language.*/);
+        super('olympusscanlation', 'Olympus Scanlation', 'https://olympusv2.gg/', Tags.Media.Manhua, Tags.Media.Manhwa, Tags.Language.Spanish);
     }
 
     public override get Icon() {
         return icon;
     }
-}
 
-// Original Source
-/*
-class OlympusScanlation extends WordPressMadara {
+    public override ValidateMangaURL(url: string): boolean {
+        return /https?:\/\/olympusv2\.gg\/series\//.test(url);
+    }
+
+    public override async FetchManga(provider: MangaPlugin, url: string): Promise<Manga> {
+        const uri = new URL(url);
+        const request = new FetchRequest(uri.href);
+        const mangaObj = await FetchWindowScript <APIManga>(request, mangaClipboardScript);
+        const slug = mangaObj.slug;
+        const mangatype = mangaObj.type;
+        const title = mangaObj.name.trim();
+        const id = JSON.stringify({ slug: slug, type: mangatype });
+        return new Manga(this,provider, id, title);
+    }
 
-    constructor() {
-        super();
-        super.id = 'olympusscanlation';
-        super.label = 'Olympus Scanlation';
-        this.tags = [ 'webtoon', 'spanish' ];
-        this.url = 'https://olympusscanlation.com';
+    public override async FetchMangas(provider: MangaPlugin): Promise<Manga[]> {
+        const mangaList = [];
+        for (let page = 1, run = true; run; page++) {
+            const mangas = await this.getMangasFromPage(page, provider);
+            mangas.length > 0 ? mangaList.push(...mangas) : run = false;
+        }
+        return mangaList;
     }
+
+    private async getMangasFromPage(page: number, provider: MangaPlugin): Promise<Manga[]> {
+        const uri = new URL(`/api/series?page=${page}&direction=asc`, this.apiUrl);
+        const request = new FetchRequest(uri.href);
+        const data = await FetchJSON<APIMangas>(request);
+        return data.data.series.data.map(element => new Manga(this, provider, JSON.stringify({ slug: element.slug, type: element.type }), element.name.trim()));
```

Comment:
IS the dynamic `type` really required, or can we just use a constant value instead?

Reply:
its needed for the api call when you fetch pages. IIRC its kinda always the same but during testing i found some others.
I would rather keep the good one than hardcoding one. Less maintenance.